checker/src/body_visitor.rs

```diff
@@ -1446,15 +1446,7 @@ impl<'analysis, 'compilation, 'tcx, E> BodyVisitor<'analysis, 'compilation, 'tcx
                         root_rustc_type,
                         move_elements,
                         |_self, path, _, new_value| {
-                            if new_value.is_bottom() || new_value.is_top() {
-                                _self.current_environment.value_map =
-                                    _self.current_environment.value_map.remove(&path);
-                                return;
-                            }
-                            _self
-                                .current_environment
-                                .value_map
-                                .insert_mut(path, new_value);
+                            _self.current_environment.update_value_at(path, new_value);
```

Comment:
Maybe also update the `insert_mut` calls to `update_value_at` in the method `attach_tag_to_elements`?

Reply:
I did have a look at it, but it does not seem to be necessary.